policies/policies.go

```diff
@@ -117,64 +147,122 @@
 	return data, nil
 }
 
-func RunMap(baseDir string, metric string, m map[string]interface{}, holder MetricConfigurationSource) (result *Result, err error) {
+type queryCache struct {
+	sync.Mutex
+	cache map[string]*rego.PreparedEvalQuery
+}
+
+func newQueryCache() *queryCache {
+	return &queryCache{
+		cache: make(map[string]*rego.PreparedEvalQuery),
+	}
+}
+
+type orElseFunc func(key string) (query *rego.PreparedEvalQuery, err error)
+
+// Get returns the prepared query for the given key. If the key was not found in the cache,
+// the orElse function is executed to populate the cache.
+func (qc *queryCache) Get(key string, orElse orElseFunc) (query *rego.PreparedEvalQuery, err error) {
 	var (
-		tx storage.Transaction
+		ok bool
 	)
 
-	// Create paths for bundle directory and utility functions file
-	bundle := fmt.Sprintf("%s/policies/bundles/%s/", baseDir, metric)
-	operators := fmt.Sprintf("%s/policies/operators.rego", baseDir)
+	// Lock the cache
+	qc.Lock()
+	// And defer the unlock
+	defer qc.Unlock()
 
-	config, err := holder.MetricConfiguration(metric)
-	if err != nil {
-		return nil, fmt.Errorf("could not fetch metric configuration: %w", err)
+	// Check, if query is contained in the cache
+	query, ok = qc.cache[key]
+	if ok {
+		return
 	}
 
-	c := map[string]interface{}{
-		"target_value": config.TargetValue.AsInterface(),
-		"operator":     config.Operator,
+	// Otherwise, the orElse function is executed to fetch the query
+	query, err = orElse(key)
+	if err != nil {
+		return nil, err
```

Comment:
```suggestion
		return
```

Reply:
I actually did that on purpose, since I have no control over what `orElse` returns and I definitely want to return `nil` here and only the error.